packages/api/src/routes/nfts-upload.js

```diff
@@ -21,78 +25,212 @@ export async function nftUpload(event, ctx) {
   const { headers } = event.request
   const contentType = headers.get('content-type') || ''
   const { user, key } = await validate(event, ctx)
-  const { db } = ctx
 
   /** @type {import('../utils/db-client-types').UploadOutput} */
   let upload
-  let sourceCid
-
   if (contentType.includes('multipart/form-data')) {
     const form = await event.request.formData()
     // Our API schema requires that all file parts be named `file` and
     // encoded as binary, which is why we can expect that each part here is
     // a file (and not a stirng).
     const files = /** @type {File[]} */ (form.getAll('file'))
 
-    const dirSize = files.reduce((total, f) => total + f.size, 0)
-    const dir = await cluster.addDirectory(files, {
-      local: dirSize > LOCAL_ADD_THRESHOLD,
+    const { car } = await packToBlob({
+      // @ts-ignore nodejs readable stream type inconsistencies
+      input: files.map((f) => ({ path: f.name, content: f.stream() })),
+      wrapWithDirectory: true,
     })
-    const { cid, size } = dir[dir.length - 1]
-
-    sourceCid = cid
-
-    upload = await db.createUpload({
-      user_id: user.id,
-      content_cid: cid,
-      source_cid: cid,
-      key_id: key?.id,
-      dag_size: size,
-      mime_type: contentType,
-      type: 'Multipart',
-      files: files.map((f) => ({
-        name: f.name,
-        type: f.type,
-      })),
+
+    upload = await uploadCar({
+      ctx,
+      user,
+      key,
+      car,
+      uploadType: 'Multipart',
+      mimeType: 'multipart/form-data',
+      files: files.map((f) => ({ name: f.name, type: f.type })),
+      structure: 'Complete',
     })
   } else {
     const blob = await event.request.blob()
     if (blob.size === 0) {
-      throw new HTTPError('Empty payload', 400)
+      throw new HTTPError('empty payload', 400)
     }
     const isCar = contentType.includes('application/car')
+    const uploadType = isCar ? 'Car' : 'Blob'
+    // If this upload is a CAR file it is not necessarily complete. i.e. it may
+    // be a chunked CAR.
+    const structure = isCar ? 'Unknown' : 'Complete'
```

Comment:
nit: Given we already to this validation below, we could use the same as car, create a let and use the conditions below.